tests/test_proxy_functional.py

```diff
@@ -3,13 +3,52 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
+import trustme
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def trustme_certs(tmp_path):
+    ca = trustme.CA()
+    ca.issue_cert("localhost")
+    ca_cert_path = tmp_path / "ca.pem"
+    ca.cert_pem.write_to_path(ca_cert_path)
+    ca_key_path = tmp_path / "server.key"
+    ca.private_key_pem.write_to_path(ca_key_path)
+    return (str(ca_cert_path), str(ca_key_path))
+
+
+@pytest.fixture
+def secure_proxy_url(trustme_certs):
+    port = proxy.common.utils.get_available_port()
+    input_args = [
+        "--num-workers",
+        "1",
+        "--threadless",
+        "--hostname",
+        "0.0.0.0",
+        "--port",
+        str(port),
+        "--cert-file",
+        trustme_certs[0],
+        "--key-file",
+        trustme_certs[1],
+    ]
+    proxy_url = "http://0.0.0.0:" + str(port)
+    with proxy.Proxy(input_args=input_args) as proxy_instance:
+        yield proxy_url
+
+
+async def test_secure_proxy_http_absolute_path(secure_proxy_url, get_request) -> None:
+    url = "https://httpbin.org/get"
```

Comment:
It's okay for a PoC but the actual tests would need to test against a local server as the regular functional client tests do.

Reply:
Got a link to a place in the tests that sets one of those up with a fixture?

Reply:
https://github.com/aio-libs/aiohttp/blob/c8e2e2c/tests/test_client_functional.py#L316

Reply:
This test is also passing, but when I try to switch it against the local server it starts failing. This is the part I really struggled with.

What I noticed is that if I switch to use the local aiohttp server as the endpoint AND I used `secure_proxy_url = "https://127.0.0.1:8899"` and configured the TLS to use certs corresponding with an externally started proxy.py with trustme certs it also works! So basically I can use proxy.py started by the tests, or an endpoint that isn't aiohttp.Server, but not both. :( Most of the code here is me switching back and forth between these approaches.